Add support for stop_signal, fixes #32858

rbjorklin opened the pull request:
### What does this PR do?

Adds stop_signal to VALID_CREATE_OPTS. This adds support for already existing functionality in docker-py.
docker-py support since 1.7.0: https://github.com/docker/docker-py/blob/master/docs/change_log.md#170
docker support since 1.9.0: https://github.com/docker/docker/releases/tag/v1.9.0
### What issues does this PR fix or reference?
#32858
### Tests written?

No
### Use case

Running systemd inside a container requires the signal SIGRTMIN+3 to shutdown gracefully, this fix makes it possible to specify the signal.

rbjorklin committed:
Add support for stop_signal, fixes #32858

This adds support for already existing functionality in docker-py.
Running systemd requires the signal SIGRTMIN+3 to shutdown properly,
this fix makes it possible to specify the signal.

rbjorklin commented:
How do I go about getting this into 2015.8 and 2016.3?

rbjorklin committed:
Adds documentation for stop_signal

rbjorklin commented:
I realized I forgot to add the documentation string but it's there now in another commit. Will I have to do another pull request?

rallytime commented:
@rbjorklin What is the difference between implementing this functionality and the `dockerng.kill` functionality that @ticosax linked in your original issue?

https://docs.saltstack.com/en/latest/ref/modules/all/salt.modules.dockerng.html#salt.modules.dockerng.kill

rbjorklin commented:
@rallytime The kill @ticosax mentioned does not either allow you to specify the signal. The default signal of `docker kill` is `SIGKILL` which basically is the equivalent of pulling the plug. This can potentially be really bad if you run an application where data consistency is important. Databases are an example of where this could potentially mean a big headache.

ticosax commented:
While `stop_signal` is probably a nice contribution. There is also [dockerng.signal](https://docs.saltstack.com/en/latest/ref/modules/all/salt.modules.dockerng.html#salt.modules.dockerng.signal) which seems to be able to serve your need.

rbjorklin commented:
@ticosax I found that while looking at the code for dockerng.running so I guess I could use that in the mean time.

@rallytime if I remove the `min_docker_py` line until the details for #32890 has been fleshed out this could be merged. How does that sound?

rallytime commented:
@rbjorklin Yes, that sounds like a good compromise. When you make that change, can you also add a `.. versionadded:: Cabron` tag in the docs for your new option? Since this is a new arg, I'd like to keep this in develop as a feature while the other PR is discussed and you can use `docker.signal` in the mean time.

rbjorklin committed:
Changed stop_signal in CREATE_OPTS

Removed min_docker_py until that functionality has been fixed.
Added a 'versionadded' tag for stop_signal.

rbjorklin commented:
I'm not sure what's happening here. I get no linting errors on my additions but there are tons of previous linting errors. Do I need to fix them to get my commit to go through? :confused:

rbjorklin committed:
Tried pleasing pylint...
